### DISCUSSION TITLE
it have serious performance issue, tested in M1 Pro(10 core) VSCode

### DISCUSSION THREAD

**Discussion by zzz6519003** [rust-lang/rust-analyzer#17806] [General]
加油~

**Comment by lnicola** [comment]
Poor performance on trait-heavy codebases like [the one](https://github.com/zzz6519003/polkadot-sdk/blob/master/polkadot/rpc/src/lib.rs#L99-L120) I assume you're testing on is a [known issue](https://github.com/rust-lang/rust-analyzer/issues/17596), and any improvements will have to wait for the new trait solver to be integrated.
